src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -312,6 +312,9 @@ private void SetCallbackHandler()
 
             Debug.Assert(_shutdownTcs.Task.IsCompleted == false);
 
+            // Stop accepting new streams after calling shutdown.
+            _acceptQueue.Writer.Complete();
```

Comment:
Is this supposed to be taken care of with the QUIC_CONNECTION_EVENT.SHUTDOWN_INITIATED_BY_TRANSPORT logic?

Reply:
I don't think so. SHUTDOWN_INITIATED_BY_TRANSPORT won't be signaled when calling close on the connection from the server. 

The other approach here is to only close it in SHUTDOWN_COMPLETE?

Reply:
Not being too familiar with this code, I'm assuming that SHUTDOWN_INITIATED_BY_TRANSPORT is for when the connection is closed by the server and SHUTDOWN_INITIATED_BY_PEER is for when the connection is closed by the client. This is from the server perspective of course. Am I wrong?

Reply:
I believe SHUTDOWN_INITIATED_BY_TRANSPORT is when there is an error in the msquic transport layer itself (like a protocol error), not server initiated shutdown.

Reply:
I see. The SHUTDOWN_COMPLETE idea sounds pretty good then. It consolidates the cleanup logic so _acceptQueue.Writer.Complete() only needs to be called in one place. I guess the downside would be the the AcceptStreamAsync() loop could take a little longer to complete.